apps/worker/src/monitor/tx.status.monitor.service.ts

```diff
@@ -34,42 +35,44 @@ export class TxStatusMonitoringService extends BaseConsumer {
       const previousKnownBlockNumber = (await this.blockchainService.getBlock(job.data.lastFinalizedBlockHash)).block.header.number.toBigInt();
       const currentFinalizedBlockNumber = await this.blockchainService.getLatestFinalizedBlockNumber();
       const blockList: bigint[] = [];
-      const blockDelay = 1 * SECONDS_PER_BLOCK * MILLISECONDS_PER_SECOND;
 
       for (let i = previousKnownBlockNumber; i <= currentFinalizedBlockNumber && i < previousKnownBlockNumber + numberBlocksToParse; i += 1n) {
         blockList.push(i);
       }
-      const txResult = await this.blockchainService.crawlBlockListForTx(job.data.txHash, blockList, [{ pallet: 'messages', event: 'MessagesStored' }]);
+      const txResult = await this.blockchainService.crawlBlockListForTx(job.data.txHash, blockList, [{ pallet: 'system', event: 'ExtrinsicSuccess' }]);
```

Comment:
we are not publishing messages.MessagesStored for every mesage but just one for every block and the better way is to just check the `system.ExtrinsicSuccess` event for success

Reply:
What if there are other ext from other apps getting included in block?
Should be make sure that event exists for the actual ext we are executing?

Reply:
But aren't we checking the extrinsic hash for that or I'm missing something?

Reply:
My understanding is that after submitting the extrinsic we get a unique txhash back. And when we are crawling the blocks we are checking that extrinsic hash for all the extrinsic inside that block. if there is a match it means it is the same extrinsic that we submitted is included in the block. `system.ExtrinsicSuccess` is enough since it is the last even that gets written for every successful extrinsic and wee don't need to check any events from messages pallet.
I'm not sure if I understand what the concern is.

Reply:
Ah i forgot the part that we are also matching the hash, yeah sounds great